pkg/providers/v2/cloud.go

```diff
@@ -61,6 +85,52 @@ type EC2Metadata interface {
 	GetMetadata(path string) (string, error)
 }
 
+func readAWSCloudConfig(config io.Reader) (*v1alpha1.AWSCloudConfig, error) {
+	if config == nil {
+		return nil, errors.New("no AWS cloud provider config file given")
+	}
+
+	// read the config file
+	data, err := ioutil.ReadAll(config)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read cloud configuration from %q [%v]", config, err)
+	}
+
+	var cfg v1alpha1.AWSCloudConfig
+	err = yaml.Unmarshal(data, &cfg)
+	if err != nil {
+		// we got an error where the decode wasn't related to a missing type
+		return nil, err
+	}
+
+	return &cfg, nil
+}
+
+// validateAWSCloudConfig validates AWSCloudConfig
+// clusterName is required
+func validateAWSCloudConfig(config *v1alpha1.AWSCloudConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if config.Kind != "AWSCloudConfig" {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("Kind"), "invalid Kind for cloud config: %q", config.Kind))
+	}
+
+	if config.APIVersion != "config.aws.io/v1alpha1" {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("APIVersion"), "invalid APIVersion for cloud config: %q", config.APIVersion))
+	}
+
+	fieldPath := field.NewPath("Config")
+	if config.Config == (v1alpha1.AWSConfig{}) {
```

Comment:
I don't think this check is required, it would fail in field-specific validation anyways